pkg/kubectl/resource/result.go

```diff
@@ -160,14 +160,22 @@ func (r *Result) Object() (runtime.Object, error) {
 	if len(versions) == 1 {
 		version = versions.List()[0]
 	}
-	return &api.List{
+	raw := []runtime.RawExtension{}
+	for _, o := range objects {
```

Comment:
should we be preferring `info.VersionedObject` in the loop above?

Reply:
I am not sure.  It looks like the `Object` is what is returned by `versions.Last()` (a runtime.VersionedObjects{}), and `VersionedObject` is what is returned by `versions.First()`.  After reading the documentation for these methods, I am not certain of which should be used here:

> 
// Last is the rightmost object in the VersionedObjects array, which is the object after
// all transformations have been applied. This is the same object that would be returned
// by Decode in a normal invocation (without VersionedObjects in the into argument).

>
// First returns the leftmost object in the VersionedObjects array, which is usually the
// object as serialized on the wire.

Reply:
... clear as mud. @smarterclayton, any guidance here? this is snipping kubectl dependencies on internal objects. Would you expect a v1.List object to contain internal or external objects?

Reply:
I think the thing that seems weird here is returning an external wrapper around internal objects

Reply:
Will these by internal objects or external objects?  This code doesn't have any dependency on internal packages, so is it even possible for it to contain internal objects?

Reply:
TestMultipleObject is ensuring it does :)

Reply:
:P  Good point.